src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/ScheduleServiceTest.kt

```diff
@@ -0,0 +1,209 @@
+  )
+
+  private val probationItems = listOf(
+    probationItem(vlbId = 1L, appId = 1L, start = nine, end = ten),
+    probationItem(vlbId = 2L, appId = 2L, start = ten, end = eleven),
+    probationItem(vlbId = 3L, appId = 3L, start = eleven, end = twelve),
+    probationItem(vlbId = 4L, appId = 4L, start = eleven, end = twelve, status = BookingStatus.CANCELLED),
+  )
+
+  private val prisonItems = listOf(
+    courtItem(vlbId = 1L, appId = 1L, start = nine, end = ten),
+    courtItem(vlbId = 2L, appId = 2L, start = ten, end = eleven),
+    probationItem(vlbId = 3L, appId = 3L, start = eleven, end = twelve),
+    probationItem(vlbId = 4L, appId = 4L, start = eleven, end = twelve, status = BookingStatus.CANCELLED),
+  )
+
+  @BeforeEach
+  fun setUpMocks() {
+    whenever(scheduleRepository.getScheduleForCourt(courtCode, LocalDate.now()))
+      .thenReturn(courtItems.filter { it.statusCode == BookingStatus.ACTIVE.name })
+
+    whenever(scheduleRepository.getScheduleForProbationTeam(probationTeamCode, LocalDate.now()))
+      .thenReturn(probationItems.filter { it.statusCode == BookingStatus.ACTIVE.name })
+
+    whenever(scheduleRepository.getScheduleForPrison(WERRINGTON, LocalDate.now()))
+      .thenReturn(prisonItems.filter { it.statusCode == BookingStatus.ACTIVE.name })
+
+    whenever(scheduleRepository.getScheduleForCourtIncludingCancelled(courtCode, LocalDate.now()))
+      .thenReturn(courtItems)
+
+    whenever(scheduleRepository.getScheduleForProbationTeamIncludingCancelled(probationTeamCode, LocalDate.now()))
+      .thenReturn(probationItems)
+
+    whenever(scheduleRepository.getScheduleForPrisonIncludingCancelled(WERRINGTON, LocalDate.now()))
+      .thenReturn(prisonItems)
+  }
+
+  @Test
+  fun `Returns the list of scheduled court items`() {
+    val response = service.getScheduleForCourt(courtCode, LocalDate.now(), false)
+    assertThat(response).isNotNull
+    assertThat(response).hasSize(3)
+    response.map { item ->
+      assertThat(item.courtCode).isEqualTo(courtCode)
+      assertThat(item.courtDescription).isEqualTo("Derby Justice Centre")
+      assertThat(item.hearingType).isEqualTo(CourtHearingType.TRIBUNAL)
+      assertThat(item.probationTeamCode).isNull()
+    }
+  }
+
+  @Test
+  fun `Returns the list of probation team items`() {
+    val response = service.getScheduleForProbationTeam(probationTeamCode, LocalDate.now(), false)
+    assertThat(response).isNotNull
+    assertThat(response).hasSize(3)
+    response.map { item ->
+      assertThat(item.probationTeamCode).isEqualTo(probationTeamCode)
+      assertThat(item.probationTeamDescription).isEqualTo("Probation")
+      assertThat(item.probationMeetingType).isEqualTo(ProbationMeetingType.PSR)
+      assertThat(item.courtCode).isNull()
+    }
+  }
+
+  @Test
+  fun `Returns the list of prison items`() {
+    val response = service.getScheduleForPrison(WERRINGTON, LocalDate.now(), false)
+    assertThat(response).isNotNull
```

Comment:
The not null checks are possibly superfluous. Checking the size straight after kinda says that anyway ...